chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go

```diff
@@ -1316,3 +1317,63 @@ func (c *ChaosExperimentHandler) validateDuplicateExperimentName(ctx context.Con
 
 	return nil
 }
+
+func (c *ChaosExperimentHandler) StopExperimentRuns(ctx context.Context, projectID string, experimentID string, experimentRunID *string, r *store.StateData) (bool, error) {
+
+	var experimentRunsID []string
+
+	tkn := ctx.Value(authorization.AuthKey).(string)
+	username, err := authorization.GetUsername(tkn)
+
+	query := bson.D{
+		{"experiment_id", experimentID},
+		{"project_id", projectID},
+		{"is_removed", false},
+	}
+	experiment, err := c.chaosExperimentOperator.GetExperiment(context.TODO(), query)
+	if err != nil {
+		return false, err
+	}
+
+	// if experimentID is provided & no expRunID is present (stop all the corresponding experiment runs)
+	if experimentRunID == nil {
+
+		// if experiment is of cron type, disable it
+		if experiment.CronSyntax != "" {
+
+			err = c.DisableCronExperiment(username, experiment, projectID, r)
+			if err != nil {
+				return false, err
+			}
+		}
```

Comment:
I can see the use case where users might want to temporarily suspend all ongoing runs but any future runs should still happen. We should give the user the option to explicitly disable the future runs imo instead of disabling by default.

Reply:
@gdsoumya This case is handled in the backend, so if a user sends a both experimentID and experimentRunID, only the current run will be stopped. but if experimentRunID is nil, it will stop all the running experiment and disable the cron if present. However, in future we are planning to add one more API to just enable/disable the cron.

Reply:
I think then in the future when the cron specific endpoint is introduced the behavior of this endpoint should be changed so that it doesn't stop the cron imo

Reply:
yes, will remove this one with the corn API